src/nile/signer.py

```diff
@@ -0,0 +1,55 @@
+"""Utility for sending signed transactions to an Account on Starknet."""
+import subprocess
+
+try:
+    from starkware.cairo.common.hash_state import compute_hash_on_elements
+    from starkware.crypto.signature.signature import private_to_stark_key, sign
+    from starkware.starknet.public.abi import get_selector_from_name
+
+    starkware_found = True
+except ImportError:
+    starkware_found = False
+
+
+class Signer:
+    """Utility for sending signed transactions to an Account on Starknet."""
+
+    def __init__(self, private_key):
+        """Construct a Signer object. Takes a private key."""
+        if not starkware_found:
+            raise Exception("starkware module not found")
+        self.private_key = private_key
+        self.public_key = private_to_stark_key(private_key)
+        self.account = None
+        self.index = 0
+
+    def sign(self, message_hash):
+        """Sign a message hash."""
+        return sign(msg_hash=message_hash, priv_key=self.private_key)
+
+    def get_nonce(self):
+        """Get the nonce for the next transaction."""
+        nonce = subprocess.check_output(
+            f"nile call account-{self.index} get_nonce", shell=True, encoding="utf-8"
```

Comment:
Here it's missing a `--network` argument so when using a non-default network this call fails